libimage/push.go

```diff
@@ -86,8 +86,8 @@ func (r *Runtime) Push(ctx context.Context, source, destination string, options
 		destRef = dockerRef
 	}
 
-	// docker-archive and only DockerV2Schema2MediaType support Gzip compression
-	if options.CompressionFormat != nil &&
+	// docker-archive and DockerV2Schema2MediaType support only Gzip compression
```

Comment:
This could benefit from a more detailed comment. Maybe
```suggestion
	// docker-archive and DockerV2Schema2MediaType support only Gzip compression.
	// But CompressionFormat can come from containers.conf, and be set to non-gzip as an universal
	// default; if that is outright not supported, we want to ignore that default.
	// OTOH ForceCompressionFormat requires CompressionFormat to be set, so don’t strip
	// gzip.
	//
	// Ideally this should all happen at a much higher layer, where the code can differentiate
	// between a value coming from containers.conf vs. the CLI.
```

Reply:
Would it be better to add a check for ForceCompressionFormat here?

If ForceCompressionFormat is set, we can assume its from the CLI, because using either `--force-compression` or `--compression-format` will set it: https://github.com/containers/podman/blob/b3c02684fdb403cd4709eadc10e0099e92c95605/cmd/podman/images/push.go#L244

If `--force-compression` is `false` but `--compression-format` is set and invalid, we can ignore it anway.

So the logic could be:
```
if <invalid combo>:
  if <force compression>:
    ERROR!
  else:
    blank out compression format
done
```

WDYT?

Reply:
That hardcodes quite a bit of undocumented knowledge about what the Podman CLI does, aesthetically I’d prefer a more explicit interface (and if we did do work to make things explicit, it seems natural to me to differentiate between default and user-specified `CompressionFormat`, making this heuristic precisely correct).

… and, practically, AFAICS it is imprecise with `--format v2s2 --compression-format zstd --force-compression=false`.

Reply:
Isn't ForceCompressionFormat the interface here? If that is set then we either use it or fail, if that isn't set then we try and revert to something that works if we can't.

Yes, that edge case would still be odd, but its odd right now.

Anyway, I'll leave it as is, it doesn't make any difference to what I need :)

Reply:
`--force-compression-format=false --format=zstd` means, down in c/image, “if the layer already exists in the destination, reuse it regardless of format; if it doesn’t exist, use zstd”.

(Admittedly the `--force-compression-format=false --format=…` behavior is not all that useful, it’s something we have for backward compatibility. So, Podman now defaults `--force-compression-format` to `true` if the user chooses a format, but it is, semantically, not just a “compression format was set” indicator.)

Using `--format v2s2 --compression-format zstd --force-compression=false` is therefore an user error, although, in effect, the user must be ready to receive a gzip-only image, so stripping the value would not be _too_ wrong.